src/deprecated.jl

```diff
@@ -105,3 +105,40 @@ function categorical!(df::DataFrame, cols::Union{Type, Nothing}=nothing;
     end
     return transform!(df, names(df, cols) .=> (x -> categorical(x, compress=compress)), renamecols=false)
 end
+
+@deprecate DataFrame(pairs::NTuple{N, Pair}; makeunique::Bool=false,
+          copycols::Bool=true) where {N} DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
+@deprecate DataFrame(columns::NTuple{N, AbstractVector}, cnames::NTuple{N, Symbol}; makeunique::Bool=false,
+          copycols::Bool=true) where {N} DataFrame(collect(AbstractVector, columns), collect(Symbol, cnames);
+              makeunique=makeunique, copycols=copycols)
+@deprecate DataFrame(columns::NTuple{N, AbstractVector}, cnames::NTuple{N, AbstractString}; makeunique::Bool=false,
+                     copycols::Bool=true) where {N} DataFrame(collect(AbstractVector, columns), [Symbol(c) for c in cnames];
+                                                              makeunique=makeunique, copycols=copycols)
+@deprecate DataFrame(columns::NTuple{N, AbstractVector};
+                     copycols::Bool=true) where {N} DataFrame(collect(AbstractVector, columns),
+                                                              gennames(length(columns)), copycols=copycols)
+@deprecate DataFrame(columns::AbstractMatrix, cnames::AbstractVector{Symbol} = gennames(size(columns, 2));
+                     makeunique::Bool=false) DataFrame(AbstractVector[columns[:, i] for i in 1:size(columns, 2)],
```

Comment:
Doesn't this work? Same below.
```suggestion
                     makeunique::Bool=false) DataFrame(collect(eachcol(columns)),
```

Reply:
This will not work on Julia 1.0 as `eachcol` is not defined there.

Reply:
Ah too bad.

Reply:
I can use `Tables.table` instead. What do you think?

Reply:
Ah yeah that's probably better for users that can't rely on Julia > 1.0.

Reply:
OK. Then we need to wait a bit till Tables.jl gets a new release. But I will fix it then.

Reply:
Compat.jl has `eachcol`

Reply:
Can we tag a new Tables.jl release now or are we waiting on a fix?

Reply:
It was tagged yesterday, so we are good to go.

I was waiting for https://github.com/queryverse/TableTraits.jl/pull/21 which is waiting for https://github.com/queryverse/TableTraits.jl/pull/21 (as discussed yesterday 😄), but since we deprecate `AbstractMatrix` constructor we do not have to wait for it now.